var/spack/repos/builtin/packages/gatetools/package.py

```diff
@@ -27,7 +28,8 @@ class Gatetools(PythonPackage):
     depends_on("py-tqdm", type=("build", "run"))
     depends_on("py-colored", type=("build", "run"))
     depends_on("py-itk@5.1.0:", type=("build", "run"))
-    depends_on("py-uproot3", type=("build", "run"))
```

Comment:
Do the older versions of gatetools still need py-uproot3?

Reply:
Nope, they've always used uproot

Reply:
The dependencies in `setup.py` list `uproot`, but version 0.9.14 needed `uproot3`. See https://github.com/spack/spack/pull/22659/files#r608006263

Unless something has changed with more recent `uproot` packages I think at least version 0.9.14 will need the `py-uproot3` dependency.

Reply:
I just checked and version 0.9.4 is the only one that will need `py-uproot3`. I also noticed that py-itk is broken with py-numpy  >= 1.20.0 but that is a separate issue.

Reply:
Gotcha. Is `py-uproot` the unversioned version, and certain versions of `py-uproot` are equivalent to `py-uproot3`? If that's the case, I would prefer to use `py-uproot@X:Y` to specify which versions were compatible. Otherwise, we can patch the setup.py to fix the wrong dependency name.

Reply:
The `py-uproot3` package contains version 3.14.4. The `py-uproot4` package contains version 4.0.2. The `py-uproot` package contains versions 4.0.6 to 5.0.3. If I recall correctly I think the `uproot3` and `uproot4` packages in PyPi, and the respective spack packages, were added for compatibility. So, I think patching `setup.py` for version 0.9.4 is the way to go.

Reply:
> I also noticed that py-itk is broken with py-numpy >= 1.20.0 but that is a separate issue.

I am still trying to pin this down. I am pretty sure that the patch needs to happen in the `itk` package but I seem to only be able to trigger it with gatetools. It is related to deprecation of `numpy.bool`. See https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

PS. I previously posted and deleted a comment that may have generated an email notification; ignore that.